src/main/java/org/jabref/gui/preferences/PreviewPreferencesTab.java

```diff
@@ -180,70 +169,50 @@ public void setValues() {
                       .onFailure(ex -> {
                           LOGGER.error("something went wrong while adding the discovered CitationStyles to the list ", ex);
                           dialogService.showErrorDialogAndWait(Localization.lang("Error adding discovered CitationStyles"), ex);
-                      }).executeWith(taskExecutor);
+                      })
+                      .executeWith(taskExecutor);
 
-        layout.setText(Globals.prefs.getPreviewPreferences().getPreviewStyle().replace("__NEWLINE__", "\n"));
+        previewTextArea.setText(previewPreferences.getPreviewStyle().replace("__NEWLINE__", "\n"));
     }
 
     @Override
     public void storeSettings() {
-        List<String> styles = new ArrayList<>();
+        PreviewPreferences previewPreferences = Globals.prefs.getPreviewPreferences();
 
         if (chosenModel.isEmpty()) {
-            chosenModel.add(Localization.lang("Preview"));
-        }
-        for (Object obj : chosenModel) {
-            if (obj instanceof CitationStyle) {
-                styles.add(((CitationStyle) obj).getFilePath());
-            } else if (obj instanceof String) {
-                styles.add("Preview");
-            }
+            chosenModel.add(previewPreferences.getTextBasedPreviewLayout());
         }
-        PreviewPreferences previewPreferences = Globals.prefs.getPreviewPreferences()
-                                                             .getBuilder()
-                                                             .withPreviewCycle(styles)
-                                                             .withPreviewStyle(layout.getText().replace("\n", "__NEWLINE__"))
-                                                             .build();
+
+        PreviewPreferences newPreviewPreferences = Globals.prefs.getPreviewPreferences()
+                                                                .getBuilder()
+                                                                .withPreviewCycle(chosenModel)
+                                                                .withPreviewStyle(previewTextArea.getText().replace("\n", "__NEWLINE__"))
+                                                                .build();
         if (!chosen.getSelectionModel().isEmpty()) {
-            previewPreferences = previewPreferences.getBuilder().withPreviewCyclePosition(chosen.getSelectionModel().getSelectedIndex()).build();
+            newPreviewPreferences = newPreviewPreferences.getBuilder().withPreviewCyclePosition(chosen.getSelectionModel().getSelectedIndex()).build();
         }
-        Globals.prefs.storePreviewPreferences(previewPreferences);
+        Globals.prefs.storePreviewPreferences(newPreviewPreferences);
 
-        // update preview
-        for (BasePanel basePanel : JabRefGUI.getMainFrame().getBasePanelList()) {
-            basePanel.getPreviewPanel().updateLayout(Globals.prefs.getPreviewPreferences());
-        }
+        // TODO: update preview
```

Comment:
Is this still "TODO"?

Reply:
Yes, at the moment we don't provide a good way for components to listen for changes in the preferences. 

At the moment the user needs to close and reopen the preview manually in order for the changes to have an effect. I think this is not too bad.

Reply:
I see...

But if only the reopening of the preview is necessary without a complete restart of JabRef... isn't it possible to close (and re-open if it was active) the preview on updating the pref settings?

Reply:
Good suggestion. The preview is now automatically closed after changes to the preferences. The user still have to reopen the preferences (but this is only relevant until the preview is integrated into the entry editor #2115 - so I don't want to invest more time into that).